auction_keeper/gas.py

```diff
@@ -38,8 +38,8 @@ def get_gas_price(self, time_elapsed: int) -> Optional[int]:
 
 
 class DynamicGasPrice(GasPrice):
-
     GWEI = 1000000000
+    failsafe_default_gas = 40
```

Comment:
I would rather this live in the configuration file and be required... what are your thoughts @EdNoepel?

There could be a scenario where gas station is down and we continue to use / start our bids @ 40... I guess logging would tell us this so any operator would be aware that the failsafe is being used...

To me what is assumed to be the baseline gas price on a TX really changes every 3 or so months now... And I don't think that pace will change. What are your thoughts about putting it in the config? any downsides?

Reply:
There is no config file, and config parameters already allow specifying a fixed gas starting price.  This handles the use case where the user didn't pass anything.  Another implementation would be to query the node for a default gas price, which would prevent us from updating it periodically.  Thoughts?

Reply:
Sorry Ed, I meant the start file. Yes we do have the `fixed gas` arg but I was thinking about adding another `fallback-fixed-gas` argument possibly?

I'm very interested in querying the node for gas price... Want to attempt that and see how it compares to etherscan? If it seems good lets implement that into our gas strat. If it seems far too low in comparison to Eth gas station lets compromise with the start file arg...

What do you think?

Reply:
I dislike the idea of a fallback arg; we'd just be abdicating responsibility to the user, and unnecessarily complicating keeper configuration.  Using node price as a fallback would eliminate `failsafe_default_gas` alltogether, and improve consistency with `pymaker`.

Reply:
A couple questions that you may have found out when you added the code that I would be curious to know:
1) Does this method / request only work with Parity nodes?
2) Have we tested this gas price comparison in relation to ethergas station... is it faster or slower and by how much? Wondering if you did some logging etc to do a basic reasoning check on what it returns.
3) is there any difficulty in obtaining this default gas price when the Ethereum network is under heavy congestion?

Reply:
1. No, it works with `geth`, `parity`, and any node supporting the JSON-RPC API.
2. All nodes provide a default gas price, and this is the value used when the user supplies no gas price.  This is the price used to submit a `pymaker` transaction when `gas_price` is not included as an arg to the `transact` method call.  And yes, this call was manually tested on kovan and mainnet.
3. There is no difficulty in obtaining the price, but it may be too low.  That's why we have a replacement mechanism and allow the user to specify an API price rather than a fixed price.